include/perfetto/public/abi/data_source_abi.h

```diff
@@ -227,6 +227,9 @@ enum PerfettoDsBufferExhaustedPolicy {
   // it will stall, retry and eventually drop data if a free chunk is not
   // acquired after a few seconds.
   PERFETTO_DS_BUFFER_EXHAUSTED_POLICY_STALL_AND_DROP = 2,
+  // Equivalent to PERFETTO_DS_BUFFER_EXHAUSTED_POLICY_DROP by default, but can
+  // be changed using the tracing configuration.
+  PERFETTO_DS_BUFFER_EXHAUSTED_POLICY_CONFIGURABLE = 3,
```

Comment:
TBH the fact that this is DROP by default but it's another entry is a bit arbitray.
IMHO would be a bit less surprising if conceptually the "configurable" is orthogonal (i.e. it's a boolean on the side, not an alternative to the policy).
I would still use PerfettoDsSetBufferExhaustedPolicy by doing

PerfettoDsSetBufferExhaustedPolicy(DROP | CONFIGURABLE)
PerfettoDsSetBufferExhaustedPolicy(STALL_AND_DROP | CONFIGURABLE)

and by making CONFIGURABLE  a high bit, like 0x80000000

Reply:
Good point.

I'm using a separate boolean instead.